### ISSUE TITLE
Add `encode`

### ISSUE THREAD

**Issue by martinsvalin** [martinsvalin/html_entities#5]
I'd like to add `encode` to the public api, like so:

```elixir
iex> HtmlEntities.encode("åäö")
"&aring;&auml;&ouml;"
```

There could be options to specify if we should use named, decimal or hex entities. The might be options to only encode unsafe characters, like < and >, and options to avoid "double-encoding" if there are already entities in the string, so "&amp;" won't become "&amp;amp;".

**Comment by DavsX** [comment]
In the case of encoding should we encode all chars > 0x255 automatically to &#XXX; format? Which HTML characters are 'unsafe' (apart from </>'"&)?

**Comment by martinsvalin** [comment]
I think `/[<>'"&]/` covers the unsafe HTML characters. (Angle brackets, single- and double-quote, ampersand. The quotes aren't really unsafe in text nodes, but can be in other contexts.)

I don't know yet if someone encoding would expect all non-ASCII characters to be encoded. I personally wouldn't want to do that, as I'd rather serve utf-8, but then I'd use the "basic" option of only encoding the unsafe characters. I always find it hard to code based on speculation… but it seems like a reasonable thing for an HTML entities encoder to encode all non-ASCII as an option.

**Comment by DavsX** [comment]
Apart from the special HTML characters the unicode characters with codepoints 160+ (the number taken from the .txt file) could be handled as such:

```elixir
  def encode(string) do
    Regex.replace(~r/[\x{00A0}-\x{FFFF}]/u, string, &encode_entity/1)
  end

  defp encode_entity(char) do
    codepoint = char |> String.to_char_list |> hd
    "&##{codepoint};"
  end
```

**Comment by jwarlander** [comment]
Encoding the unsafe characters mentioned above seems like it would be a reasonably good starting point.. If you agree, please do have a look at my pull request (#11), and let me know if there's anything you'd like to see implemented differently!

**Comment by martinsvalin** [comment]
Thank you for the PR, @jwarlander. I stalled on this because I was thinking about what an API would look like that supported different options, like encoding only unsafe characters, encoding _everything_, encoding to named entities, encoding to hex-numbered entities etc etc. 

But what I think I'd like to see is just the minimal, useful thing: to encode unsafe characters. The web has evolved far enough along that we shouldn't have to encode safe characters. Adding options on top of that for more behaviour just makes the API more complex.

So I'll consider this closed, as #11 was merged. I'll bump to 0.3.0 and release a new package to Hex.

@jwarlander, would you mind if I put you in the maintainers list? https://github.com/martinsvalin/html_entities/blob/master/mix.exs#L25

**Comment by jwarlander** [comment]
So my email response seems to have been lost.. But to summarize, I'd be fine with that - and happy to help :-)
